redis/src/types.rs

```diff
@@ -124,20 +128,83 @@ pub enum Value {
     Status(String),
     /// A status response which represents the string "OK".
     Okay,
+    /// Unordered key,value list from the server. Use `as_map_iter` function.
+    Map(Vec<(Value, Value)>),
+    /// Attribute value from the server. Client will give data instead of whole Attribute type.
+    Attribute {
+        /// Data that attributes belong to.
+        data: Box<Value>,
+        /// Key,Value list of attributes.
+        attributes: Vec<(Value, Value)>,
+    },
+    /// Unordered set value from the server.
+    Set(Vec<Value>),
+    /// A floating number response from the server.
+    Double(f64),
+    /// A boolean response from the server.
+    Boolean(bool),
+    /// First String is format and other is the string
+    VerbatimString {
+        /// Text's format type
+        format: VerbatimFormat,
+        /// Remaining string check format before using!
+        text: String,
+    },
+    /// Very large number that out of the range of the signed 64 bit numbers
+    BigNumber(BigInt),
+    /// Push data from the server.
+    Push {
+        /// Push Kind
+        kind: String,
+        /// Remaining data from push message
+        data: Vec<Value>,
+    },
+}
+
+#[derive(PartialEq, Clone, Debug)]
+pub enum VerbatimFormat {
```

Comment:
I've seen where markdown and txt were specifically mentioned in the RESP3 spec, but I wasn't clear (and haven't as of yet gone and looked at the redis source) whether that was intended as a comprehensive list or if they were just examples. Either way, I wanted to ask whether we should consider leaving this as a `String` just to avoid unnecessary breaking changes down the line if more variants are needed.

Reply:
txt is already in redis code and mkd is mentioned. I think this style of unknown and Txt, Mkd is okay.

Reply:
I discussed this with @madolson (a redis core member), and she said that these can be treated as a closed set. IMO the solution @altanozlu implemented has the best of both worlds - it's both an easy to use closed-set enum for most cases, and an open String for the edge cases of new formats, which forces the user to be aware of the fact that they received a non-standard format string.

Reply:
How about the `kind` field for `Push` types? I've found even less clarity in the docs/source as relates to this one but wanted to hear yall's thoughts on whether this might merit an enum?

Reply:
https://github.com/redis-rs/redis-rs/pull/757#discussion_r1119886733

I raised that too with @madolson. She said that the `Push` messages are more likely to be used by modules, so using a String as an open-ended format is more fitting. IIRC while modules might add new values to `VerbatimFormat`, the likelihood of using a new textual format is lesser.
IMO if you want consistency between `VerbatimString` and `Push`, I'd prefer that both be enums rather than strings. It means that matching on the common case is easier, even though it does mean that if there will be multiple modules giving multiple values for push or format the user will need to use nested match statements.

Reply:
I think `kind` could be an enum with Other(String) variant.

Reply:
Enum with Other seems pretty sane for both types to me fwiw. We really aren't sure what the "kind" of push notifications will end up existing eventually. I mentioned modules might be able to send arbitrary kind messages, but it's possible we might force modules to declare themselves as kind module + their name to be more standardized. If you have any specific thoughts I can make a note of it for future conversations.

Reply:
Thanks for the clarifications. If we were to make Push `kind` an enum as well, what would the variants be? I'm trying to remember, but I feel like I've never actually seen anything but "message" (I believe the RESP3 spec had previously mentioned others, but I see it's been updated recently). 

I'm not a huge fan of enums with a catch-all variant, but I'm willing to be overruled there, and I think it definitely seems reasonable for `VerbatimString`.

Reply:
Isn't adding new variant is major breaking change ? If it's then we need Other variant otherwise we need to bump major version every time a Redis module brings new kind to the table.

Reply:
> Isn't adding new variant is major breaking change ?

Yes, and wasn't intending for enums without a catch-all. If anything I think `String` may be preferable for `kind`. We could also consider annotating the enums with `non_exhaustive`, though I'm not sure that's desirable either.

Reply:
I'd like to implement it as enum with other kind. Btw can you give an estimation for review progress ?

Reply:
If you want to change it, go for it. If @madolson thinks the enum approach is reasonable, it works for me.

I'm almost done with the review, and overall it looks good and I don't think any major changes will be required. I think you're safe to stack additional PRs on top of this without having to worry about significant conflicts down the line. However, I'm not ready to merge this yet and want to finish up some other things before bringing in breaking changes. Thanks again for your patience -- this is definitely important work!